portfolio/src/main/webapp/script.js

```diff
@@ -13,16 +13,52 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Gets random valid index of an given array
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function getRandomIndex(array) {
+  return Math.floor(Math.random() * array.length);
+}
+
+/**
+ * Displays the text in the element in the page with given id
+ */
+function addTexttoPage(text, id){
+  const container = document.getElementById(id);
+  container.innerText = text;
+}
+
+// Global variable to track which topic has been randomly selected. 
+// Initialize to -1 to represent no topic is selected when page is first loaded. 
+var topicIndex = -1;
+const books =
+  ['Rangers', 'Exhalation', 'The Elegant Universe', 'Human Acts'];
+const tvshows =
+  ['Brookyln Nine Nine', 'Designated Survivor', 'Person of Interest', 'The Good Place', 'Broadchurch'];
+const korean =
+  ['안녕하세요!', '좋은 하루 되세요!', '오늘 날씨는 어떤가요?', '졸려요', '고맙습니다'];
+const misc =
+  ['Penguin', 'Blue', 'Baseball', 'Cheesecake'];
+const cumultative =
+  [books, tvshows, korean, misc];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+/**
+ * Generate random topic and display its description on the page 
+ */
+function getRandomTopic() {
+  const topics = ['book recommendation', 'tv show recommendation', 'korean phrase', 'thing I like'];
+
+  topicIndex = getRandomIndex(topics);
+  addTexttoPage(topics[topicIndex], 'topic-container'); 
+}
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+/**
+ * Generate random element from topic if topic is valid and display it on the page.  
+ */
+function getRandomElementfromTopic(){
+  if (topicIndex == -1){ 
```

Comment:
nit: I'm not sure how much you know about global variables and some of their pitfalls, but generally speaking mutable global variables tend to be discouraged. However, it's probably not useful to refactor at this point, we can do that later on.